crates/bevy_pbr/src/atmosphere/functions.wgsl

```diff
@@ -305,9 +309,20 @@ fn max_atmosphere_distance(r: f32, mu: f32) -> f32 {
     return mix(t_top, t_bottom, f32(hits));
 }
 
-/// Assuming y=0 is the planet ground, returns the view radius in meters
-fn view_radius() -> f32 {
-    return max(view.world_position.y * settings.scene_units_to_m, EPSILON) + atmosphere.bottom_radius;
+/// Returns the observer's position in the atmosphere
+fn get_view_position() -> vec3<f32> {
+    var world_pos = view.world_position * settings.scene_units_to_m + vec3(0.0, atmosphere.bottom_radius, 0.0);
+    
+    // If the camera is underground, clamp it to the ground surface along the local up.
+    let r = length(world_pos);
+    // add EPSILON to prevent division by zero and black pixels
```

Comment:
do you know where the division by zero is happening? i dont see any division by min_radius in this function.

Reply:
Revised the code comment above this line to better reflect why it is needed